docs/GlobalHelperFunctionsFromExternalDrivers.md

```diff
@@ -8,53 +8,50 @@ This document outlines the requirements for enabling drivers outside of `ebpfcor
 
 The current eBPF for Windows global helper function system requires:
 - **Global helpers hosted only in `ebpfcore.sys`** limiting extensibility by third-party drivers
-- **Pre-assigned numeric IDs** for all helper functions
-- **Centralized coordination** of helper ID assignments to prevent conflicts
-- **Static ID management** making it difficult for external drivers to expose custom global helper functions
+- **Pre-coordination of helper function registration** to prevent conflicts
+- **Static management** making it difficult for external drivers to expose custom global helper functions
 
 This creates barriers for independent driver development and limits the extensibility of the eBPF ecosystem by preventing external drivers from contributing global helper functions.
 
 ## Requirements
 
 ### R1: External Driver Global Helper Registration
-- Drivers outside of `ebpfcore.sys` **MUST** be able to register global helper functions as general program information providers
-- The system **MUST** support global helper function registration using descriptive names instead of pre-assigned numeric IDs
-- Registration **MUST** use existing NMR (Network Module Registrar) interfaces without modification
-- External drivers **MUST** be able to register global helpers with `helper_id = -1` to indicate name-based registration
+- Drivers outside of `ebpfcore.sys` **MUST** be able to register global helper functions
+- The system **MUST** support global helper function registration without requiring pre-coordination
+- Registration **MUST** use existing interfaces without requiring new APIs
```

Comment:
I have no objection, but why is it a requirement to not add new APIs?  Can you elaborate on what the issue is with doing so, should it turn out to be needed?

Reply:
Reworded it to make it clear that existing interfaces must remain unchanged to maintain app-compat, but new interfaces may be added if needed.